### ISSUE TITLE
Unexpected Output of Unix Timestamps

### ISSUE THREAD

**Issue by brodkin** [humanmade/Custom-Meta-Boxes#133]
Currently `datetime_unix` fields convert the received input using PHP's `strtotime` function. Unfortunately WordPress explicitly sets the default timezone as UTC which means that the dates are automatically interpreted as being in UTC even when the admin UI has been set to display dates in a different timezone. 

Being that the rest of the UI is utilizing that timezone, the expected behavior would be that dates entered would be stored with the appropriate offset. Currently retrieving the time in the user's timezone will return a different date/time than they entered.

On line 690  of classes.fields.php we currently have:

``` php
$value = strtotime( $value['date'] . ' ' . $value['time'] );
```

We could instead utilize PHP's DateTime class:

``` php
$dt = new DateTime($value['date'] . ' ' . $value['time'], new DateTimeZone(get_option('timezone_string')));
$value = $dt->format('U');
```

We would then also need to ensure that dates are also displayed to the user in their timezone when data is retrieved.

The only major issue here is that backwards compatibility would all screwed up. We'd need to introduce a `use_utc` option that would default to true or some way to ensure that people who already implemented the field under the old system would not see dates shifting. Perhaps a new `datetime_local_unix` field type instead?

I'm willing to do the work on this one, but input would be appreciated.

**Comment by swissspidy** [comment]
I know this issue has been dormant for  almost three years, but I just want to mention that storing dates without timezone information is usually a bad idea. Sure, post meta is different than a dedicated date column in a database table, so I get where this is coming from and it actually might make sense here, but generally speaking you lose important context when stripping of the timezone information for storage.

**Comment by mattheu** [comment]
I agree on this and its something I've run into before.

I believe the reason I didn't merge the pull requests before was because it would break backwards compatibility.

I recently had to do something on a project that had different events that were held in different timezones. We needed to store the date in UTC so we could sort the posts using this field, but retain the timezone for display. I whipped up this custom field, and it may be something that could solve this problem. https://github.com/mattheu/cmb-date-timezone-field
